src/Catalog/Helpers/AuditRecordHelpers.cs

```diff
@@ -0,0 +1,196 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Packaging.Core;
+using NuGet.Services.Metadata.Catalog.Persistence;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    /// <summary>
+    /// Helper methods for accessing and parsing audit records.
+    /// </summary>
+    public static class AuditRecordHelpers
+    {
+        /// <summary>
+        /// Represents the information in the audit entry for a deletion.
+        /// </summary>
+        public class DeletionAuditEntry
+        {
+            /// <summary>
+            /// Over time, we have had multiple file names for audit records for deletes.
+            /// </summary>
+            public static IList<string> FileNameSuffixes = new List<string>
+            {
+                "-Deleted.audit.v1.json",
+                "-deleted.audit.v1.json",
+                "-softdeleted.audit.v1.json",
+                "-softdelete.audit.v1.json",
+                "-delete.audit.v1.json"
+            };
+
+            public static async Task<DeletionAuditEntry> Create(Storage auditingStorage, Uri uri, CancellationToken cancellationToken, ILogger logger)
+            {
```

Comment:
Would be good to have a better name for uri to understand what uri is needed to be passed in .